src/arkode/arkode_lsrkstep.c

```diff
@@ -670,47 +409,59 @@ int lsrkStep_FullRHS(ARKodeMem ark_mem, sunrealtype t, N_Vector y, N_Vector f,
 
   /* access ARKodeLSRKStepMem structure */
   retval = lsrkStep_AccessStepMem(ark_mem, __func__, &step_mem);
-  if (retval != ARK_SUCCESS) { return (retval); }
+  if (retval != ARK_SUCCESS) { return retval; }
 
   /* perform RHS functions contingent on 'mode' argument */
   switch (mode)
   {
   case ARK_FULLRHS_START:
 
     /* compute the RHS */
-    if (!(ark_mem->fn_is_current))
+    if (!ark_mem->fn_is_current)
     {
-      retval = step_mem->fe(t, y, step_mem->Fe[0], ark_mem->user_data);
+      retval = step_mem->fe(t, y, step_mem->Fe, ark_mem->user_data);
       step_mem->nfe++;
-      if (retval != 0)
+      if (retval < 0)
       {
         arkProcessError(ark_mem, ARK_RHSFUNC_FAIL, __LINE__, __func__, __FILE__,
                         MSG_ARK_RHSFUNC_FAILED, t);
-        return (ARK_RHSFUNC_FAIL);
+        return ARK_RHSFUNC_FAIL;
+      }
+      if (retval > 0)
+      {
+        arkProcessError(ark_mem, RHSFUNC_RECVR, __LINE__, __func__, __FILE__,
+                        MSG_ARK_RHSFUNC_REPTD, t);
+        return RHSFUNC_RECVR;
       }
     }
 
     /* copy RHS vector into output */
-    N_VScale(ONE, step_mem->Fe[0], f);
+    N_VScale(ONE, step_mem->Fe, f);
 
     break;
 
   case ARK_FULLRHS_END:
     /* No further action is needed since the currently
-    available methods evaluate the RHS at the end of each time step*/
-
+	    available methods evaluate the RHS at the end of each time step*/
+    N_VScale(ONE, ark_mem->fn, f);
```

Comment:
I believe this is simply a commenting issue. Since we don’t use this function beyond the initial setup (e.g., time step size), we should adjust the comment to avoid overgeneralization. This way, it won’t imply that SSP methods compute the RHS at the new solution.

Reply:
This function can be call by other parts of the code with the `ARK_FULLRHS_END` model (e.g., for interpolation or rootfinding) in which case an in correct RHS value will be provided to those callers